src/Covenant/ASG.hs

```diff
@@ -518,6 +522,28 @@ newtype ASGBuilder (a :: Type)
     )
     via ReaderT ASGEnv (ExceptT CovenantTypeError (HashConsT Id ASGNode Identity))
 
+-- | A concrete monadic stack, providing the minimum amount of functionality
+-- needed to build an ASG using the combinators given in this module, plus
+-- some functionality needed to handle parsing an ASG from JSON.
+-- @since 1.0.0
```

Comment:
`@since` is targeting the wrong version. I also think that it's better to emphasize that this is expressly a deserialization stack - there's no reason for anyone to use it when building normally.

Is there a reason why we couldn't put this in `Covenant.JSON`?